### PR TITLE
guide for integrators

### PR DESCRIPTION
**Pull request by ievans**
# Thanks for improving Semgrep Docs 😀
Screenshot 2023-07-10 at 3 41 19 PM

### Please ensure

- [ ] A subject matter expert (SME) reviews the content
- [ ] A technical writer reviews the content or PR
- [x] This change has no security implications or else you have pinged the security team
- [x] Redirects are added if the PR changes page URLs
- [x] If you have changed any header tag links (doc/#this-kind-of-anchor), update all instances of that link

### CODE REVIEW

**Comment by s-santillan** [comment]
Thanks for contributing! We informally use [Vale](vale.sh), a natural-language linter, to help us be clear and have a consistent tone and voice in the docs. Like Semgrep, it has levels of severity; we only really care about "Blocking"-level findings. For your information here's what it looks like in the CLI:
Snag_10db113

I'll suggest some fixes, but in general we don't worry about the "warning"-level findings.

File: docs/integrating.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+description: >-
+  Learn about integrating Semgrep and the Semgrep Registry into your product.
+---
+
+# Semgrep integration guide for partners
+
+
+### Introduction
+
+We're excited that you're integrating Semgrep into your tooling! Our goal with Semgrep is to bring world-class security tools to developers based on our conviction that software will run the most exciting parts of the future. It's not something that we can do alone; we want to build a community around sharing programmatic knowledge about how to build secure software.
```
Comment (s-santillan) [inline docs/integrating.md:13]:
```suggestion
We at Semgrep, Inc. are excited that you're integrating Semgrep into your tooling! Our goal with Semgrep is to bring world-class security tools to developers based on our conviction that software will run the most exciting parts of the future. It's not something that we can do alone; we want to build a community around sharing programmatic knowledge about how to build secure software.
```

File: docs/integrating.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+slug: integrating
+append_help_link: true
+description: >-
+  Learn about integrating Semgrep and the Semgrep Registry into your product.
+---
+
+# Semgrep integration guide for partners
+
+
+### Introduction
```
Comment (s-santillan) [inline docs/integrating.md:11]:
```suggestion

```

There's no need to add "Introduction" to the top of a doc. It's an extra word users will skip over.

File: docs/integrating.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+We're excited that you're integrating Semgrep into your tooling! Our goal with Semgrep is to bring world-class security tools to developers based on our conviction that software will run the most exciting parts of the future. It's not something that we can do alone; we want to build a community around sharing programmatic knowledge about how to build secure software.
+
+
+### Requirements for Integrators
```
Comment (s-santillan) [inline docs/integrating.md:16]:
```suggestion
### Requirements for integrators
```
[Use sentence case!](https://developers.google.com/style/capitalization#:~:text=In%20document%20titles%20and%20headings,always%20capitalized%20a%20certain%20way.)

File: docs/integrating.md [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+* Do not resell rules from the registry, unless you acquire an explicit license from Semgrep, Inc. Semgrep rules are [released under the Commons Clause License](https://github.com/returntocorp/semgrep-rules/blob/develop/LICENSE), which prohibits redistribution in a commercial product.
+* State that you are using Semgrep; refer to Semgrep as capital S with the trademark: Semgrep™
+* Link to [semgrep.dev/login](https://semgrep.dev/login) to allow users to get an API token to pass to Semgrep so they can access the Pro Engine and rules.
+* Set `SEMGREP_INTEGRATION_NAME `in your environment to your domain name (e.g. "xyz.com" – this helps us reproduce & debug issues with Semgrep in your environment.
```
Comment (s-santillan) [inline docs/integrating.md:21]:
```suggestion
* Set `SEMGREP_INTEGRATION_NAME` in your environment to your domain name (for example, "xyz.com" – this helps us reproduce and debug issues with Semgrep in your environment.
```

File: docs/integrating.md [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+* Don't integrate `semgrep scan` in a CI setup. Instead use `semgrep ci`, which has diff-awareness built-in and is designed to be easy to integrate into dozens of CI environments (it's also much faster).
```
Comment (s-santillan) [inline docs/integrating.md:22]:
```suggestion
* Don't integrate `semgrep scan` in a CI setup. Instead use `semgrep ci`, which has diff-awareness built-in and is designed to be easy to integrate into dozens of CI environments. It's also much faster.
```

**Review by s-santillan — APPROVED** [review]
lgtm, minor style comments only!

File: docs/integrating.md [hunk 6]
```diff
@@ -0,0 +1,25 @@
+---
+slug: integrating
+append_help_link: true
+description: >-
+  Learn about integrating Semgrep and the Semgrep Registry into your product.
+---
+
+# Semgrep integration guide for partners
```
Comment (pabloest) [inline docs/integrating.md:8]:
```suggestion
# Semgrep integration guide
```

File: docs/integrating.md [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+
+
+## Requirements for integrators
+
+* Do not resell rules from the registry, unless you acquire an explicit license from Semgrep, Inc. Semgrep rules are [released under the Commons Clause License](https://github.com/returntocorp/semgrep-rules/blob/develop/LICENSE), which prohibits redistribution in a commercial product.
+* State that you are using Semgrep; refer to Semgrep as capital S with the trademark: Semgrep™
+* Link to [semgrep.dev/login](https://semgrep.dev/login) to allow users to get an API token to pass to Semgrep so they can access the Pro Engine and rules.
+* Set `SEMGREP_INTEGRATION_NAME` in your environment to your domain name (for example, "xyz.com"). This helps us reproduce and debug issues with Semgrep in your environment.
+* Don't integrate `semgrep scan` in a CI setup. Instead use `semgrep ci`, which has diff-awareness built-in and is designed to be easy to integrate into dozens of CI environments. It's also much faster.
+* We require enabling metrics (`--metrics=on`) by default, which lets Semgrep prioritize languages and technologies to improve speed and accuracy on.
```
Comment (pabloest) [inline docs/integrating.md:22]:
```suggestion
* Enable metrics (`--metrics=on`) by default, which lets the Semgrep team prioritize languages and technologies to improve speed and accuracy.
```

File: docs/integrating.md [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+* Contribute new public rules back to the [semgrep-rules repository](https://github.com/returntocorp/semgrep-rules). This helps us avoid community fragmentation and will automatically pull your rule into the searchable registry on semgrep.dev; plus Semgrep will maintain it for you!
+
+For more information about licensing, please refer to [this section](licensing.md) in our documentation. If you have additional questions, you can email us at [partners@semgrep.com](mailto:partners@semgrep.com).
```
Comment (pabloest) [inline docs/integrating.md:25]:
```suggestion
For more information, please refer to [the section on licensing](licensing.md) in our documentation. If you have additional questions, email us at [partners@semgrep.com](mailto:partners@semgrep.com). 
```

**Review by pabloest — APPROVED** [review]
Minor comments left via suggestions

File: docs/integrating.md (hunk 6 above)
Comment (s-santillan) [reply docs/integrating.md:8]: @pabloest Not sure if I'm overthinking it, but I'm worried that people will think "Semgrep integration guide" could mean anything - integrating Semgrep findings into Jira/DefectDojo/GHAS, etc., integrating Semgrep scans into CI, and the like.

Comment (pabloest) [reply docs/integrating.md:8]: @s-santillan Yeah I see your point. In that case, I’d suggest “Semgrep guide for integrators.” What do you think of that?

Comment (ievans) [reply docs/integrating.md:8]: I like leaving partners for people searching the docs

**Pull request merged**